src/BlockTypes/ProductSaleBadge.php

```diff
@@ -63,6 +63,68 @@ protected function get_block_type_supports() {
 		);
 	}
 
+	/**
+	 * Initialize this block.
+	 */
+	protected function initialize() {
+		parent::initialize();
+		add_filter(
+			'wp_theme_json_data_theme',
+			array( $this, 'add_default_sale_badge_styles' )
+		);
+	}
+
+	/**
+	 * Add default styles for Sale Badge block.
+	 *
+	 * @param WP_Theme_JSON_Data $theme_data The theme data object that is passed as a parameter.
+	 * @return array
+	 */
```

Comment:
Since this function is always returning the `$theme_data`, I would suggest updating the return type over here:

```suggestion
	/**
	 * Add default styles for Sale Badge block.
	 *
	 * @param \WP_Theme_JSON_Data $theme_data The theme data object that is passed as a parameter.
	 *
	 * @return \WP_Theme_JSON_Data
	 */
```

Reply:
I agree with your suggestion. I will make the necessary changes to ensure that the return type accurately reflects the data being returned by the function. Thanks 🙌🏻

Reply:
I'm curious, Is it important to include the \ before `WP_Theme_JSON_Data`?

Reply:
> I'm curious, Is it important to include the \ before WP_Theme_JSON_Data?

When you add a qualifier (the `/`) or, alternatively, an import at the beginning, you are enabling completion support for IDEs: for example, on PHPStorm, you can hover over the class name in the doc block and it will show you a description of the original class. If you `cmd + click` you are led to the file where the class was originally defined as well.